docs/quickstart.md

```diff
@@ -0,0 +1,760 @@
+Value               arn:aws:lambda:eu-central-1:123456789012:function:sam-app-HelloWorldFunction-dOcfAtYoEiGo                           
+------------------------------------------------------------------------------------------------------------------------------------------
+Successfully created/updated stack - sam-app in eu-central-1
+```
+This command will build package and deploy it to your AWS Account. Make sure you configure AWS credentials beforehand. In the outputs section you find API Url you can issue request against.
+!!! Info
+    To learn more about SAM deployment mechanism follow this [link](https://docs.aws.amazon.com/serverless-application-model/latest/developerguide/sam-cli-command-reference-sam-deploy.html).
+
+## API Gateway router
+Let's say you have been asked to create another method that respond with caller name under `/hello/<name>` URL path.
+One approach would be to create another lambda with required method and configure API gateway to call it.
+=== "app_name.py"
+
+    ```python
+    def hello_name(name):
+        return {"statusCode": 200, "body": {"message": f"hello {name}!"}}
+
+
+    def lambda_handler(event, context):
+        name = event["pathParameters"]["name"]
+        return hello_name(name)
+    ```
+
+=== "template.yaml"
+
+    ```yaml hl_lines="22-33"
+    AWSTemplateFormatVersion: "2010-09-09"
+    Transform: AWS::Serverless-2016-10-31
+    Description: > 
+        Sample SAM Template for powertools-quickstart
+    Globals:
+        Function:
+            Timeout: 3
+    Resources:
+        HelloWorldFunction:
+            Type: AWS::Serverless::Function
+            Properties:
+            CodeUri: hello_world/
+            Handler: app.lambda_handler
+            Runtime: python3.9
+            Events:
+                HelloWorld:
+                Type: Api 
+                Properties:
+                    Path: /hello
+                    Method: get
+        HelloWorldFunctionName:
+            Type: AWS::Serverless::Function
+            Properties:
+            CodeUri: hello_world/
+            Handler: app_name.lambda_handler
+            Runtime: python3.9
+            Events:
+                HelloWorldName:
+                Type: Api
+                Properties:
+                    Path: /hello/{name}
+                    Method: get
+    Outputs:
+        HelloWorldApi:
+            Description: "API Gateway endpoint URL for Prod stage for Hello World function"
+            Value: !Sub "https://${ServerlessRestApi}.execute-api.${AWS::Region}.amazonaws.com/Prod/hello/"
+    ```
+
+This way definitively works. The downside of this approach is that you need to create new lambda file for every method or path and also configure SAM template to add new lambda configuration and configure API Gateway to point into them. For small additional methods it might be too much boilerplate code.  
```

Comment:
This sound a bit biased, where we present a downside without having a problem or complexity. Maybe first describe the situation when we need to add a new method, and that it will be too much boilerplate for many URL paths?

Reply:
rewrote to:
`This way certainly works for simple use-case. But what happens if your application gets bigger and we need to cover numerous URL paths and HTTP methods for them? If that is the case, we should:
* Add a new lambda handler with business logic for each new URL path and HTTP method used.
* Add a new Lambda configuration to a SAM template file to map the lambda function to the required path and HTTP URL method.
This could result in a number of similar lambda files and large SAM configuration file with similar configuration sections. 
In this case where we see that the addition of new URL paths lead to the boilerplate code, we should lean towards the routing approach.`